### PR TITLE
[DET-2271] feat: how-to use trained model checkpoints guide

### PR DESCRIPTION
**Pull request by sidneyw**
I re-forked the repository. This is a continuation of https://github.com/determined-ai/determined/pull/197

### CODE REVIEW

File: docs/how-to/use-trained-models.txt [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+Querying Checkpoints
+--------------------
+
+The :class:`ExperimentReference <determined.experimental.ExperimentReference>`
+class is a reference to an experiment. It is retrieved via the
+:class:`Determined <determined.experimental.Determined>` class. The reference
+contains the ``top_checkpoint`` method. Without arguments, the method will
+check the experiment configuration searcher field for the ``metric`` and
+``smaller_is_better`` values. These values are used to sort the experiments
```
Comment (vishnu2kmohan) [inline docs/how-to/use-trained-models.txt:23]: Nit: Apostrophe required - `experiment's`?

File: docs/how-to/use-trained-models.txt [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+checkpoints by validation performance. The following is a snippet from an
```
Comment (vishnu2kmohan) [inline docs/how-to/use-trained-models.txt:24]: Misleading. It's not the snippet itself, but the searcher configuration in the snippet. How about:
The `searcher:` settings in the following snippet from an experiment configuration file...

File: docs/how-to/use-trained-models.txt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+experiment configuration file will result in checkpoints being sorted by the
+loss metric in ascending order.
+
+.. code:: yaml
+
+   searcher:
+     metric: "loss"
+     smaller_is_better: true
+
+The following snippet can be run with any python interpreter after the
```
Comment (vishnu2kmohan) [inline docs/how-to/use-trained-models.txt:34]: Suggestion: `any python (3.6 or later) interpreter`? Or just `a python interpreter`?
I'm fairly certain I've not been successful with some implicit dependencies if I'm on python 3.8 (e.g., for older versions of TensorFlow, 1.14, IIRC).

File: docs/how-to/use-trained-models.txt [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+.. code:: python
+
+  from determined.experimental import Determined
+
+  checkpoints = Determined.get_experiment(id).top_n_checkpoints(5)
+
+
+The :class:`TrialReference <determined.experimental.TrialReference>` is used for
+more fine grained control over checkpoint selection. It contains a
```
Comment (vishnu2kmohan) [inline docs/how-to/use-trained-models.txt:70]: Nit: `more` feels redundant, given we're stating it's used for `fine grained control`, anyways.

File: docs/how-to/use-trained-models.txt [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+.. code:: python
+
+  from determined.experimental import Determined
+
+  checkpoint = Determined().get_experiment(id).top_checkpoint()
+  checkpoint_path = checkpoint.download()
+
+The ``load`` method of the :class:`Checkpoint
+<determined.experimental.Checkpoint>` class downloads the checkpoint if it
```
Comment (vishnu2kmohan) [inline docs/how-to/use-trained-models.txt:118]: Suggestion, could use a couple of oxford commas.
`downloads the checkpoint, if it does not exist, and then...`

File: docs/how-to/use-trained-models.txt [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+does not exist on locally and then load it into memory in a python
```
Comment (vishnu2kmohan) [inline docs/how-to/use-trained-models.txt:119]: `s/load/loads/g`

File: docs/how-to/use-trained-models.txt [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+process as shown in the following snippet.
```
Comment (vishnu2kmohan) [inline docs/how-to/use-trained-models.txt:120]: Suggestion: Another oxford comma - `process, as shown in ...`

File: docs/how-to/use-trained-models.txt [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+  checkpoint = Determined.get_experiment(id).top_checkpoint()
+  model = checkpoint.load()
+
+Tensorflow checkpoints are saved in the ``saved_model`` format and are loaded
+as trackable objects (see documentation for `tf.compat.v1.saved_model.load_v2 <https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/saved_model/load_v2>`_ for details).
+
+PyTorch checkpoints are saved via pickle and loaded as ``nn.Module``
+(see `pytorch serialization <https://pytorch.org/docs/stable/notes/serialization.html>`_ for details).
+
+Download Checkpoints via the CLI
```
Comment (vishnu2kmohan) [inline docs/how-to/use-trained-models.txt:135]: Are we missing docs for `det experiment download <exp_id>`? 
Or is that follow-on work that we decided not to land yet?

Comment (sidneyw) [reply docs/how-to/use-trained-models.txt:135]: This hasn't landed yet. I am planning on opening the PR for it today. I will update the docs accordingly in that PR.

File: docs/how-to/use-trained-models.txt [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+=========================
+
+Determined provides APIs for downloading trained checkpoints and loading them
+into memory in a python process.
+
+The following guide will discuss:
+
+#. Querying trained model checkpoints from trials and experiments.
+#. Loading models into memory in a python process.
+#. Using the determined CLI to download checkpoints to disk.
```
Comment (katport) [inline docs/how-to/use-trained-models.txt:13]: capitalize Determined

File: docs/how-to/use-trained-models.txt [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+the checkpoints in ascending or descending order with respect to the
+``sort_by`` metric.
+
+.. code:: python
+
+  from determined.experimental import Determined
+
+  checkpoint = Determined().get_experiment(id).top_checkpoint(sort_by="accuracy", smaller_is_better=False)
+
+You may also query the top n checkpoints to receive a sorted list of
```
Comment (katport) [inline docs/how-to/use-trained-models.txt:56]: capitalize N since it's not in quotes?

File: docs/how-to/use-trained-models.txt [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+
+The following snippet downloads a checkpoint from persistent storage to the
+local file system.
+
+.. code:: python
+
+  from determined.experimental import Determined
+
+  checkpoint = Determined().get_experiment(id).top_checkpoint()
+  checkpoint_path = checkpoint.download()
```
Comment (katport) [inline docs/how-to/use-trained-models.txt:115]: If possible, maybe add that they can state the path to download to?

Comment (sidneyw) [reply docs/how-to/use-trained-models.txt:117]: Done

**Pull request merged**